Tests/test_file_jpeg.py

```diff
@@ -88,6 +88,32 @@ def test_app(self):
             assert im.info["comment"] == b"File written by Adobe Photoshop\xa8 4.0\x00"
             assert im.app["COM"] == im.info["comment"]
 
+    @pytest.mark.parametrize(
+        "keep_rgb, no_default_app_segments, expect_app0, expect_app14",
+        (
+            (False, False, True, False),
+            (True, False, False, True),
+            (False, True, False, False),
+            (True, True, False, False),
+        ),
+    )
+    def test_default_app_write(
+        self,
+        keep_rgb,
+        no_default_app_segments,
+        expect_app0,
+        expect_app14,
+    ):
+        out = BytesIO()
+        hopper().save(
+            out,
+            format="JPEG",
+            keep_rgb=keep_rgb,
+            no_default_app_segments=no_default_app_segments,
+        )
+        assert (b"\xff\xe0" in out.getvalue()) == expect_app0
```

Comment:
This is not reliable check. Can we open image instead and check `im.applist` instead?